src/routers/WormholeL2VoteAggregatorRouter.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import {L2VoteAggregator} from "src/L2VoteAggregator.sol";
+import {WormholeL2VoteAggregator} from "src/WormholeL2VoteAggregator.sol";
+
+contract WormholeL2VoteAggregatorRouter is WormholeL2VoteAggregator {
```

Comment:
Should we have an optimized version like this or incorporate these into the default `L2VoteAggregator`? If yes should we do the same thing for the `l1Unlock` function in the `WormholeL2ERC20`?

Reply:
I like the idea of having it as an optional extension rather than part of the core contract so this way seems good 👍 

I do wonder about the name "Router" though. Typically I think of a Router as a contract that exists outside some core protocol contracts and coordinates common interactions with those core contracts (think Uniswap router in periphery vs. AMM code in core).

The name therefore seems appropriate for some of our other calldata optimization work, where we were in fact wrapping around core contracts that we didn't even control under the hood. In this case, though, we're extending the core contracts and interacting with their internals. This leads to two questions:

1. Is Router the best name, and if not, what would be more appropriate? `OptimizedWormholeL2VoteAggregator`. `WormholeL2VoteAggregatorOptimizor`. I dunno, but Router feels possibly misleading
2. Why did you choose the extension pattern rather than writing them as more traditional, external routers? I'm guessing it's because we needed `msg.sender` to be the voter? Not saying we should change, just want to make sure I know what drove the design decision.

Reply:
As for the question of whether to do it for the ERC20 unlock method: let's hold off for now. It's not something that would be used regularly by a user taking advantage of the product, so it doesn't seem worth it.

Reply:
On the design decision you are right, it was because we use `msg.sender`, and on the name I agree it is misleading. What do you think about `WormholeL2VoteAggregatorFallback`? I think optimized works, but it still seems a little vague.

Reply:
For me "Fallback" doesn't provide much information. I like all 3 of @wildmolasses' suggestions and would be ok with any of them. Ed has a reputation for being good at naming!

Reply:
Nice, I will go with `WormholeL2VoteAggregatorCalldataCompressor`

Reply:
https://github.com/ScopeLift/l2-flexible-voting/pull/69/commits/818f5c0f2e84648483255533a82db58bffcd69bd

Reply:
Lgtm, can you rename the file as well @alexkeating

Reply:
I thought the file was renamed which one should I rename?